From 34a0afeb251d14c2c98e8b61a85f6621a9ffe3d0 Mon Sep 17 00:00:00 2001
From: Haavard Skinnemoen <hskinnemoen@atmel.com>
Date: Tue, 6 Nov 2007 19:38:24 +0100
Subject: [PATCH] opcontrol: don't use kill -s

Busybox's implementation of "kill" doesn't understand the "-s SIG"
option. Use "-SIG" instead.

Signed-off-by: Haavard Skinnemoen <hskinnemoen@atmel.com>
---
 utils/opcontrol |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/utils/opcontrol b/utils/opcontrol
index 7cb68a7..5a75cd2 100644
--- a/utils/opcontrol
+++ b/utils/opcontrol
@@ -908,7 +908,7 @@ do_stop()
 		return
 	fi
 
-	kill -s 0 `cat $LOCK_FILE` 2>/dev/null
+	kill -0 `cat $LOCK_FILE` 2>/dev/null
 	if test "$?" -ne 0; then
 		echo "Detected stale lock file. Removing." >&2
 		rm -f "$LOCK_FILE"
@@ -919,7 +919,7 @@ do_stop()
 		echo "Stopping profiling."
 		echo 0 >/dev/oprofile/enable
 	fi
-	kill -s USR2 `cat $LOCK_FILE` 2>/dev/null
+	kill -USR2 `cat $LOCK_FILE` 2>/dev/null
 }
 
 
@@ -932,7 +932,7 @@ do_kill_daemon()
 		return
 	fi
 
-	kill -s 0 `cat $LOCK_FILE` 2>/dev/null
+	kill -0 `cat $LOCK_FILE` 2>/dev/null
 	if test "$?" -ne 0; then
 		echo "Detected stale lock file. Removing." >&2
 		rm -f "$LOCK_FILE"
@@ -1274,7 +1274,7 @@ do_start_daemon()
 {
  
 	if test -f "$LOCK_FILE"; then
-		kill -s 0 `cat $LOCK_FILE` 2>/dev/null
+		kill -0 `cat $LOCK_FILE` 2>/dev/null
 		if test "$?" -eq 0; then
 			return;
 		else
@@ -1341,7 +1341,7 @@ do_start()
 	if test "$KERNEL_SUPPORT" = "yes"; then
 		echo 1 >$MOUNT/enable
 	fi
-	kill -s USR1 `cat $LOCK_FILE` 2>/dev/null
+	kill -USR1 `cat $LOCK_FILE` 2>/dev/null
 	echo "Profiler running."
 }
 
-- 
1.5.3.4

